app/src/main/java/org/mozilla/focus/telemetry/ActivationPing.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.telemetry
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.focus.GleanMetrics.Activation
+import org.mozilla.focus.GleanMetrics.Pings
+
+/**
+ * Ensures that only one activation ping is ever sent.
+ *
+ * (Taken from Fenix)
+ */
+class ActivationPing(private val context: Context) {
+
+    private val prefs: SharedPreferences by lazy {
+        context.getSharedPreferences(
+            "${this.javaClass.canonicalName}.prefs", Context.MODE_PRIVATE)
```

Comment:
I prefer explicit names: Code gets moved, renamed, and suddenly we read a different preference and send a bunch of pings again. :)

Reply:
Oh, that's a good point, I didn't think about that!